src/Uno.UI/Microsoft/UI/Xaml/Controls/PullToRefresh/RefreshContainer/Native/NativeRefreshControl.Android.cs

```diff
@@ -0,0 +1,293 @@
+				//}
+
+				desiredChildSize = MeasureChild(child, scrollSpace);
+
+				// Give opportunity to the the content to define the viewport size itself
+				(child as ICustomScrollInfo)?.ApplyViewport(ref desiredChildSize);
+			}
+
+			return desiredChildSize;
+		}
+
+		protected override Size ArrangeOverride(Size slotSize)
+		{
+			var child = RefreshControl.Content;
+
+			if (child != null)
+			{
+				var desiredChildSize = LayoutInformation.GetDesiredSize(child);
+
+				var occludedPadding = new Thickness(0);
+				slotSize.Width -= occludedPadding.Left + occludedPadding.Right;
+				slotSize.Height -= occludedPadding.Top + occludedPadding.Bottom;
+
+				var width = Math.Max(slotSize.Width, desiredChildSize.Width);
+				var height = Math.Max(slotSize.Height, desiredChildSize.Height);
+
+				ArrangeChild(child, new Rect(
+					0,
+					0,
+					width,
+					height
+				));
+
+				//ScrollContentPresenter.ScrollOwner?.TryApplyPendingScrollTo();
+
+				// Give opportunity to the the content to define the viewport size itself
+				(child as ICustomScrollInfo)?.ApplyViewport(ref slotSize);
+
+			}
+
+			return slotSize;
+		}
+
+		protected override string Name => Panel.Name;
+	}
+
+	public override bool OnInterceptTouchEvent(MotionEvent e)
+	{
+		_lastOnInterceptTouchEvent = base.OnInterceptTouchEvent(e);
+		switch (e.Action)
+		{
+			case MotionEventActions.Down:
+				_isSwiping = false;
+				_ignoreGesture = false;
+				_gestureStart = new System.Drawing.PointF(e.GetX(), e.GetY());
+				break;
+			case MotionEventActions.Move:
+				return ShouldInterceptMove(e);
+			default:
+				_isSwiping = false;
+				_ignoreGesture = false;
+				break;
+		}
+
+		return _lastOnInterceptTouchEvent;
+	}
+
+	private bool ShouldInterceptMove(MotionEvent e)
+	{
+		//If the Swipe already started swiping we let it handle the touch until it is completed.
+		if (_isSwiping)
+		{
+			return true;
+		}
+		else
+		{
+			Vector2 cumulativeDistance = new Vector2(
+											Math.Abs(e.GetX() - _gestureStart.X),
+											Math.Abs(e.GetY() - _gestureStart.Y)
+										);
+
+			//Here we validate that the touch is not horizontal. 
+			//This is to ensure that it will not override horizontal touches for children such as flipviews.
+			if (_ignoreGesture || cumulativeDistance.X > _touchSlop)
+			{
+				_ignoreGesture = true;
+				return false;
+			}
+
+			if (cumulativeDistance.Y > _touchSlop && // touchSlop is the minimal distance for us to determine it is a scroll
+				cumulativeDistance.Y > cumulativeDistance.X && // Check the scroll is vertical
+				e.GetY() - _gestureStart.Y > 0 && // Check if the scroll is positive so that we don't intercept the scrolling of the AVP/List
+				!CanChildScrollUp()) // Validate if the child can scroll up so that we only allow swipe refresh when the content is scrolled.
+			{
+				_isSwiping = true;
+
+				return true;
+			}
+			else
+			{
+				return false;
+			}
+		}
+	}
+
+	public override bool OnTouchEvent(MotionEvent e)
```

Comment:
Sounds really bad for me to override those methods. Why you didn't use the multi targeted manipulation events?

Reply:
Turns out these were not needed in the end.